src/database/translate-operations.ts

```diff
@@ -0,0 +1,373 @@
+                    });
+                  })
+              );
+
+              return prisma.chainConfiguration.create({
+                data: {
+                  chain: {
+                    connect: await getId(key),
+                  },
+                  active,
+                  sponsor,
+                  topUpWallet: { connect: topUpWalletResults },
+                  airSeekerConfig: airseekerConfig,
+                },
+                select: { id: true },
+              });
+            })
+          );
+
+          await prisma.beacon.create({
+            data: {
+              name,
+              templateId,
+              description,
+              id: beaconId,
+              providerName: provider.apiMetadata.name,
+              chains: { connect: createdChainConfig },
+            },
+          });
+
+          (
+            await Promise.all(
+              Object.entries(provider.deployments).flatMap(([key, value]) =>
+                Promise.all(
+                  Object.entries(value.airnode).flatMap(([appType, nextLevelDown]) =>
+                    Promise.all(
+                      Object.entries(nextLevelDown)
+                        .filter(([_cloudType, airnodeNested]) => airnodeNested?.ois)
+                        .flatMap(async ([cloudType, airnodeNestedSource]) => {
+                          const airnodeNested = airnodeNestedSource as Config;
+
+                          await Promise.all(
+                            airnodeNested.ois.map(async (ois) => {
+                              const result = await prisma.oIS.create({
+                                data: {
+                                  provider: { connect: { name: createdProvider.name } },
+                                  title: ois.title,
+                                  version: ois.version,
+                                  ois: airnodeNested.ois,
+                                },
+                                select: { id: true },
+                              });
+                              return { id: result.id };
+                            })
+                          );
+
+                          const deployedAt = new Date(
+                            parseInt(`20${key.substring(0, 2)}`),
+                            parseInt(key.substring(2, 4)) - 1,
+                            parseInt(key.substring(4, 6)),
+                            parseInt(key.substring(7, 9)),
+                            parseInt(key.substring(9, 11))
+                          );
+
+                          return {
+                            applicationType:
+                              appType === 'airnode' ? ApplicationType.Airnode : ApplicationType.Airseeker,
+                            appVersion: airnodeNested.nodeSettings.nodeVersion,
+                            deployedAt,
+                            cloudType: cloudType.toLowerCase() === 'GCP' ? CloudType.GCP : CloudType.AWS,
```

Comment:
Should be toUpperCase?

Reply:
Yes, wow, thanks. The if isn't actually needed with `toUpperCase` :D